torch/_dynamo/compiled_autograd.py

```diff
@@ -271,51 +271,31 @@ def set_node_origin(self, node_name, node_index):
 
 compiled_autograd_enabled = False
 
-# We may have code like:
-# with enable(compiler_fn):
-#   ...
-#   with disable():
-#     ...
-#   ...
-# The disable() call just want to disable compiled autograd temporarily.
-# But overall the feature is enabled.
-#
-# The code covered by the disable context manager has no way to know if
-# compiled autograd is overall eanbled. Use another variable
-# compiled_autograd_enabled_count to indicate how many times compiled
-# autograd has been enabled in the call stack for this purpose.
-compiled_autograd_enabled_count = 0
-
 
 @contextlib.contextmanager
 def enable(compiler_fn):
+    global compiled_autograd_enabled
+    assert not compiled_autograd_enabled
+    compiled_autograd_enabled = True
     prior = torch._C._dynamo.compiled_autograd.set_autograd_compiler(
         functools.partial(AutogradCompilerInstance, compiler_fn)
     )
     torch._C._dynamo.compiled_autograd.set_verbose_logging(
         snapshot_verbose_logging_enabled()
     )
-    global compiled_autograd_enabled, compiled_autograd_enabled_count
-    compiled_autograd_enabled = True
-    compiled_autograd_enabled_count += 1
     try:
         with torch.autograd.set_multithreading_enabled(False):
             yield
     finally:
-        compiled_autograd_enabled_count -= 1
         if not prior:
             compiled_autograd_enabled = False
         torch._C._dynamo.compiled_autograd.set_autograd_compiler(prior)
 
 
 @contextlib.contextmanager
-def disable():
+def disable_compiler():
```

Comment:
Why the rename here?  Should the enable/disable APIs match for consistentcy?

Reply:
keeping it as disable is confusing since this no longer affects the compiled_autograd_enabled flag. i'll just rename both to {enable,disable}_compiler since the enable/disable API is changing to the torch._dynamo.config flag